### PR TITLE
New functionality, refreshes the ListView to change data in prop onFetch

### PR DESCRIPTION
**Pull request by debian789**
Update the view if a parameter is changed in callback in props onFetch

This allows you to create a search engine that implements a ListView

```

constructor (props) {
  super(props)
  this.state = {
    searchText: 'Hi',
    forceUpdate: false
  }
}


_onFetch(textSerch, page = 1, callback, options) {
  client.newPappers( {search: textSearch, page: page}, function (err, data) {
    if (!err) {
       callback(data.results);
    }
  })
 }

 onSearchChange (event) {
   var textSearch = event.nativeEvent.text.toLowerCase()
   this.setState({searchText: textSearch, forceUpdate: true})
 }

render () {
  return (<View>
<SearchBar
          onSearchChange={this.onSearchChange.bind(this)}
          isLoading={this.state.isLoading}
          onFocus={() =>
             this.refs.listview2 && this.refs.listview2.getScrollResponder().scrollTo(0, 0)}
       />
<GiftedListView
    rowView= {this._renderRowView.bind(this)}
    onFetch= {this._onFetch.bind(this, this.props.textSearch)}
    firstLoader={true} // display a loader for the first fetching
    pagination={true} // enable infinite scrolling using touch to load more
    refreshable={true} // enable pull-to-refresh for iOS and touch-to-refresh for Android
    withSections={false} // enable sections
    forceUpdate={this.props.forceUpdate}
    customStyles={{
      paginationView: {
        backgroundColor: 'red',
        marginBottom: 50
     },
    }}
    refreshableTintColor="blue"
    /></View>)
}
```

### CODE REVIEW

**Comment by skyFi** [comment]
+1

**Review by nxtpool — APPROVED** [review]

File: GiftedListView.js [hunk 1]
```diff
@@ -200,6 +202,16 @@ var GiftedListView = React.createClass({
     this.props.onFetch(this._getPage(), this._postRefresh, {firstLoad: true});
   },
 
+  componentWillReceiveProps() {
```
Comment (K0den) [inline GiftedListView.js]: Change this to `componentWillReceiveProps(nextProps)`

File: GiftedListView.js [hunk 2]
```diff
@@ -200,6 +202,16 @@ var GiftedListView = React.createClass({
...
+    if (this.props.forceUpdate) {
```
Comment (K0den) [inline GiftedListView.js]: And then change this to `nextProps.forceUpdate` otherwise you need two cycles before it updates

**Review by K0den — CHANGES_REQUESTED** [review]
Use nextProps params. Have you tried to run your example? I haven't but you are switching between using this.state and this.props so I don't think it'll work

File: GiftedListView.js (hunk 1 above)
Comment (debian789) [reply GiftedListView.js]: update with nextProps

File: GiftedListView.js (hunk 2 above)
Comment (debian789) [reply GiftedListView.js]: change with nextProps

**Review by K0den — APPROVED** [review]

**Comment by K0den** [comment]
Then we just need @FaridSafi to wake up or give someone else write access :)

**Review by sjafri5 — COMMENTED** [review]
when I was running this change I was getting a lot of glitchiness ( the load circle would almost come up and then wouldnt) because of the setState call on line 207. This means even when there's a minor state change the load bar will began to pull up. I remove lines 207-209 and that solved the issue. Otherwise works great.

**Pull request merged**